src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php

```diff
@@ -52,8 +52,8 @@ class RedirectToPreferredLocaleListener
     public function __construct(UrlGeneratorInterface $urlGenerator, $locales, $defaultLocale = null)
     {
         $this->urlGenerator = $urlGenerator;
+        $this->locales = array_filter(explode('|', trim($locales)));
```

Comment:
why exactly the `array_filter`? Maybe just `$this->locales = explode('|', trim($locales, ' |'));`? Seems unnecessary to iterate the full array.